fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaViewModel.java

```diff
@@ -284,20 +297,109 @@ public void sendChirp(String text, @Nullable MessageID parentId, long timestamp)
+      String channel = laoChannel + SOCIAL + token.getPublicKey().getEncoded();
+      Log.d(TAG, PUBLISH_MESSAGE);
+      MessageGeneral msg = new MessageGeneral(token, deleteChirp, mGson);
+
+      Disposable disposable =
+          mLaoRepository
+              .sendPublish(channel, msg)
+              .subscribeOn(Schedulers.io())
+              .observeOn(AndroidSchedulers.mainThread())
+              .timeout(5, TimeUnit.SECONDS)
+              .subscribe(
+                  answer -> {
+                    if (answer instanceof Result) {
+                      Log.d(TAG, "Deleted chirp with messageId: " + msg.getMessageId());
+                      Toast.makeText(
+                              getApplication().getApplicationContext(),
+                              "Deleted chirp!",
+                              Toast.LENGTH_LONG)
+                          .show();
+                    } else {
+                      Log.d(TAG, "Failed to delete chirp");
+                      Toast.makeText(
+                              getApplication().getApplicationContext(),
+                              "Error when deleting chirp. Please try again",
+                              Toast.LENGTH_LONG)
+                          .show();
+                    }
+                  },
+                  throwable ->
+                      Log.d(TAG, "Timed out waiting for result on chirp/delete", throwable));
+      disposables.add(disposable);
+    } catch (KeyException e) {
+      Log.e(TAG, getApplication().getString(R.string.error_pop_token), e);
+      Toast.makeText(
+              getApplication().getApplicationContext(),
+              R.string.error_toast_pop_token + e.getMessage(),
+              Toast.LENGTH_LONG)
+          .show();
+    }
+  }
+
+  public List<MessageID> getChirpIdList(String laoId) {
+    Lao lao = mLaoRepository.getLaoByChannel(ROOT + laoId);
+    return lao.getChirpsIdInOrder();
+  }
+
+  public Map<MessageID, Chirp> getAllChirps(String laoId) {
+    Lao lao = mLaoRepository.getLaoByChannel(ROOT + laoId);
+    return lao.getAllChirps();
+  }
+
+  /**
+   * Check whether the sender of a chirp is the current user
+   *
+   * @param sender String of the PoPToken PublicKey
+   * @return true if the sender is the current user
+   */
+  public boolean isOwner(String sender) {
+    Log.d(TAG, "Testing if the sender is also the owner");
+    boolean isOwner = false;
+    String laoChannel = ROOT + getLaoId().getValue();
+    LAOState laoState = mLaoRepository.getLaoById().get(laoChannel);
+    if (laoState == null) {
+      Log.e(TAG, LAO_FAILURE_MESSAGE);
+      return false;
+    }
+
+    try {
+      PoPToken token = mKeyManager.getValidPoPToken(laoState.getLao());
+      isOwner = sender.equals(token.getPublicKey().getEncoded());
+    } catch (KeyException e) {
+      Log.e(TAG, getApplication().getString(R.string.error_pop_token), e);
     }
+    return isOwner;
```

Comment:
I like to create variables to simplify code.
But there, dropping it will make the code easier to read.

```suggestion
    String laoChannel = ROOT + getLaoId().getValue();
    LAOState laoState = mLaoRepository.getLaoById().get(laoChannel);
    if (laoState == null) {
      Log.e(TAG, LAO_FAILURE_MESSAGE);
      return false;
    }

    try {
      PoPToken token = mKeyManager.getValidPoPToken(laoState.getLao());
      return sender.equals(token.getPublicKey().getEncoded());
    } catch (KeyException e) {
      Log.e(TAG, getApplication().getString(R.string.error_pop_token), e);
      return false;
    }
```